### DISCUSSION TITLE
Deployment Checklist Documentation

### DISCUSSION THREAD

**Discussion by pragyanshrestha-ibriz** [icon-project/icon-bridge#359] [Ideas]
This is a draft checklist that will be converted to a markdown documentation after incorporating suggestions from different teams.

**Comment by bbist** [comment]
Pre-Deployment Checklist:
1. Make sure that receiver is able to work with mainnet at expected throughput.

Mainnet Initial Deployment Checklist:
1. Setup the system to be used for deployment, pre-install all the dependencies required by the deployment script.
    1. Generate keystores to be used as god wallet for deployment
    2. Fund generated god wallets to have some minimum balance required for full deployment
2. Use dedicated RPC endpoints for deployment. Also, try to make deployment script as stable as possible.
3. Always start with testnet deployment and verify methods are working.
4. After mainnet deployment, verify all readonly methods of each contract are showing expected values.
5. Remove deployer (god wallet) as owner.
6. Fund mainnet BMR wallets.
7. AWS secrets manager configuration on ICONLoop server. Send Ajaya's video.
8. Setup monitoring on BMR instance before running BMR.
9. Add slack webhook url on BMR config. Make sure that BMR configuration like sync concurrency and other parameters are appropriately set.
10. Investigate system resource utilization after BMR deployment, netstat (active connections to RPC endpoints, CPU, RAM utilization, Disk Usage, etc)
11. Git tagging/versioning of code after each deployment and upgrade.
12. Publish mainnet deployment details on github for Nexus/Marketing Team, contract addresses and network details.
13. Run full smoke and e2e tests on testnet, followed by smoke tests on mainnet.
14. Don't remove the deployment code after deployment, need it for later upgrade. 

Mainnet BMR Upgrade:
1. 


Mainnet Contract Upgrade:
1. 


Mainnet Code Review Session:
1.

**Reply by CyrusVorwald** [reply]
Note that this is the steps performed to deploy the BSC integration for the beta. The checklist is a work in progress that will be added to over time. The list for full integration launch should also include:

Code coverage requirements passed and accepted (to be written)
Static analysis requirements passed and accepted (to be written)
List of functional tests passed and accepted (to be written)
List of smoke tests passed and accepted (see https://github.com/icon-project/icon-bridge/discussions/314, still needs to be converted to a .md file)
Documentation criteria accepted (see https://github.com/icon-project/community/blob/main/guidelines/technical/software-development-guidelines.md#documentation)
External security audit

**Comment by andrii-kl** [comment]
Just few additional cases

Considering that we now have more than one pair of chains(bsc, near, snow).

We should check:
1) The relay is working properly with several pairs of chains.
2) Transfers between not icon chain are working.

**Reply by CyrusVorwald** [reply]
Now that SNOW and NEAR are on testnet, have these been tested among all chains?
